base/libgit2/callbacks.jl

```diff
@@ -144,8 +144,8 @@ function authenticate_userpass(creds::UserPasswordCredentials, libgit2credptr::P
     isusedcreds = checkused!(creds)
 
     if creds.prompt_if_incorrect
-        username = creds.user
-        userpass = creds.pass
+        username = Base.get(ENV, "AUTH_USER", creds.user)
+        userpass = Base.get(ENV, "AUTH_PASS", creds.pass)
```

Comment:
I've found some examples of these being used in other applications. Mostly I was following the naming conventions set by `authenticate_ssh` which uses `SSL_KEY_PATH`, `SSH_PUB_KEY_PATH` and `SSH_KEY_PASS`. I'm fine with making these Julia specific: `JULIA_AUTH_USER` and `JULIA_AUTH_PASS`.